pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -748,6 +754,27 @@ func (r *RepoSyncReconciler) validateNamespaceName(namespaceName string) error {
 	return nil
 }
 
+func (r *RepoSyncReconciler) validateCACertSecret(ctx context.Context, repoSync *v1beta1.RepoSync) error {
```

Comment:
If we change the signature to accept either `Git` or `SecretReference` (and `namespace`) rather than a `RootSync`/`RepoSync`, we can move this declaration to `reconcilerBase`. This would be preferable because then we only need to define this logic in one place (rather than duplicated for RootSync/RepoSync)

Reply:
refactored function to reconciler_base. Updated to take secretRef string and namespace param